XiEditor/MarkerBar.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class Marker {
+    //let description: String
+    let line: Int
+    let color: NSColor
+
+    init(_ line: Int, color: NSColor) {
+        self.line = line
+        self.color = color
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class MarkerBar: NSView, CALayerDelegate {
+    var markers: [Marker] = []
+    var backgroundColor: NSColor = NSColor.clear
+    var parent: EditViewController!
+
+    var markerBarWidth: CGFloat = 16.0 {
```

Comment:
If this is 15.0 or smaller then some weird rendering issues with the scroll bars happen